lib/dfTools.class.php

```diff
@@ -1376,4 +1376,65 @@ public static function preg_replace($pattern, $replacement, $subject)
     {
         return is_null($subject) ? null : preg_replace($pattern, $replacement, $subject);
     }
+
+    public static function get_min_variant_prices($products, $include_taxes)
+    {
+        $min_prices_by_product_id = [];
+        foreach ($products as $product) {
+            if (self::is_parent($product)) {
+                continue;
+            }
+
+            $product_id = $product['id_product'];
+            $variant_id = $product['id_product_attribute'];
+            $variant_price = self::get_price($product_id, $include_taxes, $variant_id);
+            $variant_onsale_price = self::get_onsale_price($product_id, $include_taxes, $variant_id);
+
+            if (key_exists($product_id, $min_prices_by_product_id)) {
+                $current_min_prices = $min_prices_by_product_id[$product_id];
+
+                // Even though, in order to track the minimum, we can only focus on
+                // the sales price, we still need both prices of the variant
+                // in order to properly and consistenly populate the price of
+                // the parent to show the proper price vs sale_price when searching
+                // in the layer
+                if ($variant_onsale_price < $current_min_prices['onsale_price']) {
+                    $min_prices_by_product_id[$product_id]['price'] = $variant_price;
+                    $min_prices_by_product_id[$product_id]['onsale_price'] = $variant_onsale_price;
+                }
+            } else {
+                $min_prices_by_product_id[$product_id] = ['price' => $variant_price, 'onsale_price' => $variant_onsale_price];
+            }
+        }
+
+        return $min_prices_by_product_id;
+    }
+
+    public static function is_parent($product)
+    {
+        return isset($product['id_product_attribute']) && (int) $product['id_product_attribute'] === 0;
```

Comment:
Although the id_product_attribute should be an integer, casting any string to int would also produce 0 as a result:

![image](https://github.com/doofinder/doofinder-prestashop/assets/128705267/4958fb7b-c29b-40df-a63e-2ba02cf969f3)

Suggestion:

```suggestion
        return isset($product['id_product_attribute']) && is_numeric($product['id_product_attribute']) && (int) $product['id_product_attribute'] === 0;
```

Reply:
Hecho! Muchas gracias por el apunte!